Source/common/SNTBlockMessage.m

```diff
@@ -123,35 +127,76 @@ + (NSString *)stringFromHTML:(NSString *)html {
 }
 
 + (NSString *)replaceFormatString:(NSString *)str
-                         withDict:(NSDictionary<NSString *, NSString * (^)()> *)replacements {
+                         withDict:(NSDictionary<NSString *, NSString *> *)replacements {
   __block NSString *formatStr = str;
 
-  [replacements
-    enumerateKeysAndObjectsUsingBlock:^(NSString *key, NSString * (^computeValue)(), BOOL *stop) {
-      NSString *value = computeValue();
-      if (value) {
-        formatStr = [formatStr stringByReplacingOccurrencesOfString:key withString:value];
-      }
-    }];
+  [replacements enumerateKeysAndObjectsUsingBlock:^(NSString *key, NSString *value, BOOL *stop) {
+    if ((id)value != [NSNull null]) {
+      formatStr = [formatStr stringByReplacingOccurrencesOfString:key withString:value];
+    }
+  }];
 
   return formatStr;
 }
 
-// Returns either the generated URL for the passed in event, or an NSURL from the passed in custom
-// URL string. If the custom URL string is the string "null", nil will be returned. If no custom
-// URL is passed and there is no configured EventDetailURL template, nil will be returned.
-// The following "format strings" will be replaced in the URL, if they are present:
 //
```

Comment:
Nit: Bit of an awkward comment now with only the replacement strings listed. Maybe add a small description?